Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -78,14 +76,17 @@ public struct Configuration: Equatable {
         }
         self.disabledRules = validDisabledRules
 
-        self.rules = rules.filter {
-            !validDisabledRules.contains($0.dynamicType.description.identifier)
+        self.rules = rules.filter { rule in
+            let id = rule.dynamicType.description.identifier
+            if validDisabledRules.contains(id) { return false }
+            return enabledRules.contains(id) || !(rule is OptInRule)
```

Comment:
I'd prefer not to have the `OptInRule` or possibly a different mechanism for it. It makes `enablesRules` only useful for opting in to non-default rules, rather a definitive whitelist.